robottelo/ui/session.py

```diff
@@ -46,3 +50,21 @@ def logout(self):
     def close(self):
         """Exits session and also closes the browser (used in shell)"""
         self.browser.close()
+
+    def get_org_name(self):
+        """
+        Make a Organization and cache its name to be returned through session,
+        avoiding overhead of its recreation on each test.
+
+        Organization Must be at same state (not mutate) at the end of the test
+
+        Create your own organization if mutation is needed. Otherwise other
+        tests can break with your tests side effects
+
+        :return: str: Organization name
+        """
+        if 'org_name' in _org_cache:
+            return _org_cache['org_name']
+        _org_cache['org_name'] = gen_string('alpha')
+        make_org(self, org_name=_org_cache['org_name'])
```

Comment:
put the org in cache after it gets really created eg: inversed
```python
org_name = gen_string('alpha')
make_org(self, org_name=org_name)
_org_cache['org_name'] = org_name
```

in case of a failure in make_org, we still have a chance to create it, but this way, the org name will be in the cache but the org may not exist